### ISSUE TITLE
Integer Overflow exception when loading X-Ray. No error state for page (keeps "loading" forever)

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#5960]
Unable to use "exact" X-rays at all. Seeing this error:

```clojure
09-13 13:56:15 ERROR metabase.middleware :: GET /api/x-ray/segment/76 500 (2 mins) (17 DB calls)
{:message "integer overflow",
 :stacktrace
 ["feature_extraction.core$dataset__GT_features.invokeStatic(core.clj:29)"
  "feature_extraction.core$dataset__GT_features.invoke(core.clj:26)"
  "feature_extraction.core$eval51364$fn__51365.invoke(core.clj:104)"
  "api.x_ray$fn__51423$fn__51424.invoke(x_ray.clj:65)"
  "api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:248)"
  "api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:243)"
  "api.x_ray$fn__51423.invokeStatic(x_ray.clj:58)"
  "api.x_ray$fn__51423.invoke(x_ray.clj:58)"
  "middleware$enforce_authentication$fn__42275.invoke(middleware.clj:122)"
  "api.routes$fn__51503.invokeStatic(routes.clj:65)"
  "api.routes$fn__51503.invoke(routes.clj:65)"
  "routes$fn__51575$fn__51576.doInvoke(routes.clj:75)"
  "routes$fn__51575.invokeStatic(routes.clj:71)"
  "routes$fn__51575.invoke(routes.clj:71)"
  "middleware$log_api_call$fn__42388$fn__42390.invoke(middleware.clj:330)"
  "middleware$log_api_call$fn__42388.invoke(middleware.clj:329)"
  "middleware$add_security_headers$fn__42324.invoke(middleware.clj:245)"
  "middleware$bind_current_user$fn__42279.invoke(middleware.clj:142)"
  "middleware$maybe_set_site_url$fn__42328.invoke(middleware.clj:268)"]}
```
Steps to repro:

1. Add the baseball stats DB
2. Create a new segment for the "Games" table called "Games since 1970". The definition is `game_dt` (Game Date) is greater than `19700000`
3. Try to view "exact" X-ray for the Segment. API fails with 500 after a few minutes.
4. Error message is never displayed; loading spinner stays up forever

We should definitely look at why this bug is happening and also add somewhere to show error messages here instead of leaving the user to just assume the page is insanely slow

**Comment by sbelak** [comment]
The overflow part is fixed with #5946. Should probably do something more informative and/or graceful on the frontend though.

**Comment by salsakran** [comment]
what in #5946 can be merged? It has a WIP on it and is failing tests ...

**Comment by sbelak** [comment]
Wasn't really sure until when do we keep WIP. All of it as far as I'm concerned. 

It's failing some js stuff. I've merged in the latest master & 0.26.0 and pushed. Should be green now.

**Comment by salsakran** [comment]
It's a WIP until it's ready to be reviewed. Passing linters/tests is a bare minimum

**Comment by Globegitter** [comment]
@sbelak I am just testing latest master and getting an integer overflow error. Not sure if related but here is the stack trace:

```
09-15 11:04:55 ERROR sync.util :: Error generating fingerprint for Field 188 'id': integer overflow
("sync.analyze.fingerprint.number$fn__29922$average__29927$fn__29928.invoke(number.clj:9)"
 "sync.analyze.fingerprint.number$fn__29922$average__29927.invoke(number.clj:6)"
 "sync.analyze.fingerprint.number$fn__29945$number_fingerprint__29950$fn__29951.invoke(number.clj:17)"
 "sync.analyze.fingerprint.number$fn__29945$number_fingerprint__29950.invoke(number.clj:12)"
 "sync.analyze.fingerprint$fn__30179$type_specific_fingerprint__30184$fn__30185.invoke(fingerprint.clj:26)"
 "sync.analyze.fingerprint$fn__30179$type_specific_fingerprint__30184.invoke(fingerprint.clj:21)"
 "sync.analyze.fingerprint$fn__30207$fingerprint__30212$fn__30213.invoke(fingerprint.clj:35)"
 "sync.analyze.fingerprint$fn__30207$fingerprint__30212.invoke(fingerprint.clj:29)"
 "sync.analyze.fingerprint$fn__30258$fingerprint_table_BANG___30263$fn__30264$fn__30270.invoke(fingerprint.clj:57)"
 "sync.util$do_with_error_handling.invokeStatic(util.clj:115)"
 "sync.util$do_with_error_handling.invoke(util.clj:110)"
 "sync.analyze.fingerprint$fn__30258$fingerprint_table_BANG___30263$fn__30264.invoke(fingerprint.clj:56)"
 "sync.analyze.fingerprint$fn__30258$fingerprint_table_BANG___30263.invoke(fingerprint.clj:52)"
 "sync.analyze.fingerprint$fn__30430$fingerprint_fields_BANG___30435$fn__30436.invoke(fingerprint.clj:156)"
 "sync.analyze.fingerprint$fn__30430$fingerprint_fields_BANG___30435.invoke(fingerprint.clj:152)"
 "sync.analyze$fn__30559$analyze_table_BANG___30564$fn__30565.invoke(analyze.clj:72)"
 "sync.analyze$fn__30559$analyze_table_BANG___30564.invoke(analyze.clj:68)"
 "sync.analyze$fn__30582$analyze_db_BANG___30587$fn__30588$fn__30589.invoke(analyze.clj:86)"
 "sync.util$do_with_error_handling.invokeStatic(util.clj:115)"
 "sync.util$do_with_error_handling.invoke(util.clj:110)"
 "sync.util$do_with_error_handling.invokeStatic(util.clj:113)"
 "sync.util$do_with_error_handling.invoke(util.clj:110)"
 "driver$fn__22936.invokeStatic(driver.clj:209)"
 "driver$fn__22936.invoke(driver.clj:209)"
 "driver$fn__22910$G__22691__22919.invoke(driver.clj:34)"
 "sync.util$sync_in_context$fn__29435.invoke(util.clj:106)"
 "sync.util$with_db_logging_disabled$fn__29432.invoke(util.clj:98)"
 "sync.util$with_start_and_finish_logging$fn__29427.invoke(util.clj:87)"
 "sync.util$with_sync_events$fn__29424.invoke(util.clj:70)"
 "sync.util$with_duplicate_ops_prevented$fn__29415.invoke(util.clj:49)"
 "sync.util$do_sync_operation.invokeStatic(util.clj:133)"
 "sync.util$do_sync_operation.invoke(util.clj:130)"
 "sync.analyze$fn__30582$analyze_db_BANG___30587$fn__30588.invoke(analyze.clj:82)"
 "sync.analyze$fn__30582$analyze_db_BANG___30587.invoke(analyze.clj:77)"
 "sync$fn__31886$sync_database_BANG___31891$fn__31892$fn__31893.invoke(sync.clj:30)"
 "sync.util$do_with_error_handling.invokeStatic(util.clj:115)"
 "sync.util$do_with_error_handling.invoke(util.clj:110)"
 "sync.util$do_with_error_handling.invokeStatic(util.clj:113)"
 "sync.util$do_with_error_handling.invoke(util.clj:110)"
 "driver$fn__22936.invokeStatic(driver.clj:209)"
 "driver$fn__22936.invoke(driver.clj:209)"
 "driver$fn__22910$G__22691__22919.invoke(driver.clj:34)"
 "sync.util$sync_in_context$fn__29435.invoke(util.clj:106)"
 "sync.util$with_db_logging_disabled$fn__29432.invoke(util.clj:98)"
 "sync.util$with_start_and_finish_logging$fn__29427.invoke(util.clj:87)"
 "sync.util$with_sync_events$fn__29424.invoke(util.clj:70)"
 "sync.util$with_duplicate_ops_prevented$fn__29415.invoke(util.clj:49)"
 "sync.util$do_sync_operation.invokeStatic(util.clj:133)"
 "sync.util$do_sync_operation.invoke(util.clj:130)"
 "sync$fn__31886$sync_database_BANG___31891$fn__31892.invoke(sync.clj:26)"
 "sync$fn__31886$sync_database_BANG___31891.invoke(sync.clj:19)"
 "events.sync_database$process_sync_database_event$fn__31932.invoke(sync_database.clj:34)")
```

**Comment by sbelak** [comment]
It's not, but fixed in #5982. Thanks!

**Comment by salsakran** [comment]
fixed via #5982
